src/main/flight/pid.c

```diff
@@ -967,9 +967,13 @@ void FAST_CODE pidController(float dT)
 
         // Limit desired rate to something gyro can measure reliably
         pidState[axis].rateTarget = constrainf(rateTarget, -GYRO_SATURATION_LIMIT, +GYRO_SATURATION_LIMIT);
+
 #ifdef USE_GYRO_KALMAN
-        gyroKalmanSetSetpoint(axis, pidState[axis].rateTarget);
+        if (gyroConfig()->kalmanEnabled) {
```

Comment:
I this this should be moved to `pidProfile()` since it's technically not gyro anymore, but rather kalman pre-filtering.